File: apps/watch/src/main/kotlin/io/deathproject/watch/core/DecisionEngine.kt
package io.deathproject.watch.core

class DecisionEngine {
    fun decide(state: WatchState): Decision {
        if (state.snoozeUntil != null && state.now.isBefore(state.snoozeUntil)) {
            return Decision.Skip(SkipReason.SNOOZED)
        }
        if (state.isSleeping) {
            return Decision.Skip(SkipReason.SLEEPING)
        }
        if (state.isCharging) {
            return Decision.Skip(SkipReason.CHARGING)
        }
        return Decision.SendAlert
    }
}

File: apps/watch/Sources/WatchCore/DecisionEngine.swift
import Foundation

public final class DecisionEngine {
    public init() {}

    public func decide(_ state: WatchState) -> Decision {
        if let snoozeUntil = state.snoozeUntil, state.now < snoozeUntil {
            return .skip(.snoozed)
        }
        if state.isSleeping {
            return .skip(.sleeping)
        }
        if state.isCharging {
            return .skip(.charging)
        }
        return .sendAlert
    }
}
